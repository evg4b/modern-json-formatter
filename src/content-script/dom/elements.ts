import { element } from './helpres';

export const toggle = () => element({ class: 'toggle' });
export const ellipsis = () => element({ class: 'ellipsis' });
export const comma = () => element({ content: ',', class: 'comma' });
export const colon = () => element({ content: ':', class: 'colon' });
export const bracket = {
  open: () => element({ content: '{', class: 'bracket bracket-open' }),
  close: () => element({ content: '}', class: 'bracket bracket-close' }),
};

export const squareBracket = {
  open: () => element({ content: '[', class: 'bracket square-bracket-open' }),
  close: () => element({ content: ']', class: 'bracket square-bracket-close' }),
};

export const propertiesCount = (count: number) =>
  element({
    content: `// ${count} propert${count === 1 ? 'y' : 'ies'}`,
    class: 'properties-count',
  });

export const itemsCount = (count: number) =>
  element({
    content: `// ${count} item${count === 1 ? '' : 's'}`,
    class: 'items-count',
  });
