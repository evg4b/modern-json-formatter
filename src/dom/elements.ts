import { element } from './helpres';

export const toggle = () => element({ class: 'toggle' });
export const ellipsis = () => element({ class: 'ellipsis' });
export const comma = () => element({ content: ',', class: 'comma' });
export const colon = () => element({ content: ':', class: 'colon' });
