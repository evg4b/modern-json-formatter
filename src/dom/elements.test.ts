import { bracket, colon, comma, ellipsis, itemsCount, propertiesCount, squareBracket, toggle } from './elements';

describe('elements', () => {
  test('toggle', () => {
    const element = toggle();
    expect(element).toMatchSnapshot();
  });

  test('ellipsis', () => {
    const element = ellipsis();
    expect(element).toMatchSnapshot();
  });

  test('comma', () => {
    const element = comma();
    expect(element).toMatchSnapshot();
  });

  test('colon', () => {
    const element = colon();
    expect(element).toMatchSnapshot();
  });

  describe('square-bracket', () => {
    test('open', () => {
      const element = squareBracket.open();
      expect(element).toMatchSnapshot();
    });

    test('close', () => {
      const element = squareBracket.close();
      expect(element).toMatchSnapshot();
    });
  });

  describe('bracket', () => {
    test('open', () => {
      const element = bracket.open();
      expect(element).toMatchSnapshot();
    });

    test('close', () => {
      const element = bracket.close();
      expect(element).toMatchSnapshot();
    });
  });

  test('propertiesCount', () => {
    const element = propertiesCount(5);
    expect(element).toMatchSnapshot();
  })

  test('itemsCount', () => {
    const element = itemsCount(5);
    expect(element).toMatchSnapshot();
  })
});
