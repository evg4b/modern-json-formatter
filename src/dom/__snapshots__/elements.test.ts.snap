// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`elements bracket close 1`] = `
<span
  class="bracket bracket-close"
>
  }
</span>
`;

exports[`elements bracket open 1`] = `
<span
  class="bracket bracket-open"
>
  {
</span>
`;

exports[`elements colon 1`] = `
<span
  class="colon"
>
  :
</span>
`;

exports[`elements comma 1`] = `
<span
  class="comma"
>
  ,
</span>
`;

exports[`elements ellipsis 1`] = `
<span
  class="ellipsis"
/>
`;

exports[`elements itemsCount 1`] = `
<span
  class="items-count"
>
  // 5 items
</span>
`;

exports[`elements propertiesCount 1`] = `
<span
  class="properties-count"
>
  // 5 properties
</span>
`;

exports[`elements square-bracket close 1`] = `
<span
  class="bracket square-bracket-close"
>
  ]
</span>
`;

exports[`elements square-bracket open 1`] = `
<span
  class="bracket square-bracket-open"
>
  [
</span>
`;

exports[`elements toggle 1`] = `
<span
  class="toggle"
/>
`;
